src/fermilib/utils/_grid.py

```diff
@@ -0,0 +1,48 @@
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+from __future__ import absolute_import
+
+import itertools
+
+
+class Grid:
+    """
+    A multi-dimensional grid of points.
+    """
+
+    def __init__(self, n_dimensions, grid_length, length_scale):
+        """
+        Args:
+            n_dimensions (int): The number of dimensions the grid lives in.
+            grid_length (int): The number of points along each grid axis.
+            length_scale (float): The distance between adjacent grid points.
+        """
```

Comment:
I personally agree, but type checks are unusual for python code so I don't think it's pythonic to actually add the checks. There is a general expectation that duck typing should just work.